test/extensions/filters/network/common/fuzz/uber_filter.cc

```diff
@@ -0,0 +1,194 @@
+              dynamic_cast<Filters::Common::ExtAuthz::GrpcClientImpl*>(&callbacks);
+          const std::string empty_body{};
+          const auto expected_headers =
+              Filters::Common::ExtAuthz::TestCommon::makeHeaderValueOption({});
+          auto check_response = Filters::Common::ExtAuthz::TestCommon::makeCheckResponse(
+              Grpc::Status::WellKnownGrpcStatus::Ok, envoy::type::v3::OK, empty_body,
+              expected_headers);
+          // Give response to the grpc_client by calling onSuccess().
+          grpc_client_impl->onSuccess(std::move(check_response), span_);
+          return async_request_.get();
+        })));
+
+    EXPECT_CALL(*async_client_factory_, create()).WillOnce(Invoke([&] {
+      return std::move(async_client_);
+    }));
+
+    EXPECT_CALL(factory_context_.cluster_manager_.async_client_manager_,
+                factoryForGrpcService(_, _, _))
+        .WillOnce(Invoke([&](const envoy::config::core::v3::GrpcService&, Stats::Scope&, bool) {
+          return std::move(async_client_factory_);
+        }));
+  }
+}
+void UberFilterFuzzer::fuzzerSetup() {
+  // Setup process when this fuzzer object is constructed.
+  // For a static fuzzer, this will only be executed once.
+
+  // Get the pointer of read_filter when the read_filter is being added to connection_.
+  read_filter_callbacks_ = std::make_shared<NiceMock<Network::MockReadFilterCallbacks>>();
+  ON_CALL(read_filter_callbacks_->connection_, addReadFilter(_))
+      .WillByDefault(Invoke([&](Network::ReadFilterSharedPtr read_filter) -> void {
+        read_filter_ = read_filter;
+        read_filter_->initializeReadFilterCallbacks(*read_filter_callbacks_);
+      }));
+  // Prepare sni for sni_cluster filter and sni_dynamic_forward_proxy filter.
+  ON_CALL(read_filter_callbacks_->connection_, requestedServerName())
+      .WillByDefault(testing::Return("fake_cluster"));
+  // Prepare time source for filters such as local_ratelimit filter.
+  factory_context_.prepareSimulatedSystemTime();
+  // Prepare address for filters such as ext_authz filter.
+  addr_ = std::make_shared<Network::Address::PipeInstance>("/test/test.sock");
+  read_filter_callbacks_->connection_.remote_address_ = addr_;
+  read_filter_callbacks_->connection_.local_address_ = addr_;
+  async_request_ = std::make_unique<Grpc::MockAsyncRequest>();
+}
+
+UberFilterFuzzer::UberFilterFuzzer() : time_source_(factory_context_.simulatedTimeSystem()) {
+  fuzzerSetup();
+}
+bool UberFilterFuzzer::invalidInputForFuzzer(const std::string& filter_name,
+                                             Protobuf::Message* config_message) {
+  // System calls such as reading files are prohibited in this fuzzer. Some input that crashes the
+  // mock/fake objects are also prohibited.
+  const std::string name = Extensions::NetworkFilters::Common::FilterNameUtil::canonicalFilterName(
+      std::string(filter_name));
+  if (filter_name == NetworkFilterNames::get().DirectResponse) {
```

Comment:
What do you think about separating out these (and the per filter setup) into a separate source file?
Also OK with leaving as TODO and iterating. Eventually this file will be longer and longer :)

Reply:
> What do you think about separating out these (and the per filter setup) into a separate source file?
> Also OK with leaving as TODO and iterating. Eventually this file will be longer and longer :)

That makes sense! I have separated them.